documentation/MIGRATION-guide-for-next-generation-management-libraries.md

```diff
@@ -214,15 +224,113 @@ In next-generation, we keep this feature except
+| cancel the operation                                                       | N/A                                          | `cancelOperation()`                                                                                                                                                                                             |
+| get the current operation state                                            | `getPollState()`                             | `getOperationState()`                                                                                                                                                                                           |
+| access the final result                                                    | `getOperationResponse()`                     | `getResult()`                                                                                                                                                                                                   |
+| serialize the poller state                                                 | N/A                                          | `toString()`                                                                                                                                                                                                    |
+| get the most recent response                                               | `getMostRecentResponse()`                    | can be accessed using the [`onResponse`](https://github.com/Azure/azure-sdk-for-js/blob/35f86a68def5022d4c03a32d4c8c0362a0158772/sdk/core/core-client/src/interfaces.ts#L115) callback in the operation options |
+| get the current operation status                                           | `getOperationStatus()`                       | can be accessed using the [`onResponse`](https://github.com/Azure/azure-sdk-for-js/blob/35f86a68def5022d4c03a32d4c8c0362a0158772/sdk/core/core-client/src/interfaces.ts#L115) callback in the operation options |
+
+And here are examples of how to commonly use one of the `beginCreateOrUpdateAndWait` LROs found in `@azure/compute`:
+
+<!-- markdownlint-disable MD033 -->
+<table>
+  <tr>
+    <td colspan="2">
+      <p>
+        Operations that returns a poller.
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <th>Current Libraries</th>
+    <th>Next Generation</th>
+  </tr>
+  <tr>
+    <td>
+      <pre lang="typescript">
+const computeClient = new compute.ComputeManagementClient(credential, subscriptionId);
+const poller = await computeClient.dedicatedHosts.beginCreateOrUpdate(
+  resourceGroupName,
+  hostGroupName,
+  hostName,
+  parameter
+);
+console.log(`Are we done yet? ${poller.getPollState().state !== "InProgress"}`)
```

Comment:
Just out of curiousity. Does this `InProgress` work for all kinds of ARM Polling strategy such as async operation polling or location header polling ?

Reply:
I am not sure sure how the old poller works tbh, I just found `InProgress` a valid value of the `state` enum so I used it but you're right, there could be differences. I switched it to show the current status.